### ISSUE TITLE
Monte Carlo Dropout during inference

### ISSUE THREAD

**Issue by ssuralcmu** [ultralytics/ultralytics#17780]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am trying to add dropout during inference to Yolov8 or Yolov11 for uncertainty estimation. What is the best way to achieve this?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ssuralcmu, thank you for your interest in Ultralytics 🚀! We recommend consulting our documentation for new users, which includes a wealth of information on various usage examples. You might find answers to common questions there.

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and ensure you are following our tips for achieving the best training results.

Join the Ultralytics community where it suits you best. For real-time chat, head to Discord 🎧. Prefer in-depth discussions? Check out Discourse. Or dive into threads on our Subreddit to share knowledge with the community.

## Upgrade

It's always a good idea to upgrade to the latest `ultralytics` package along with all required dependencies to verify your issue is not already resolved in the latest version. Try the following command:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python, and PyTorch preinstalled):

- Notebooks with free GPU: Gradient, Colab, and Kaggle
- Google Cloud Deep Learning VM
- Amazon Deep Learning AMI
- Docker Image

## Status

If the CI badge is green, all Ultralytics CI tests are currently passing. CI tests verify the correct operation of all YOLO modes and tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but an Ultralytics engineer will also assist you soon. 😊

**Comment by glenn-jocher** [comment]
@ssuralcmu to implement Monte Carlo Dropout during inference with YOLOv8 or YOLO11, you can modify the model's architecture to keep dropout layers active in evaluation mode by setting their `training=True` during the forward pass. This may involve subclassing the model and overriding the `forward` method. Note that this is an advanced custom modification, and caution should be exercised to ensure correct implementation.
